aptos-move/framework/aptos-stdlib/sources/table.move

```diff
@@ -96,6 +106,41 @@ module aptos_std::table {
     // ======================================================================================================
     // Internal API
 
+    struct NewTableEvent has drop, store {
+        handle: u128,
+        key_type: String,
+        value_type: String,
+    }
+
+    struct TableMetadata has key, store {
+        new_table_events: EventHandle<NewTableEvent>,
+    }
+
+    public fun initialize(account: &signer) {
+        assert!(address_of(account) == @aptos_std, error::permission_denied(ENOT_STD_ACCOUNT));
+
+        assert!(!exists<TableMetadata>(@aptos_std), error::already_exists(ETABLE_METADATA_ALREADY_EXISTS));
+        move_to(account, TableMetadata {
+            new_table_events: new_event_handle<NewTableEvent>(account)
+        });
+    }
+
+    /// Emits NewTableEvent if the event handle is installed under @aptos_std.
+    fun maybe_emit_create_table_event<K: copy + drop, V>(handle: u128) acquires TableMetadata {
+        // In unit tests the event handle is not installed.
```

Comment:
We shouldn't make a special case just for unit tests. Can't unit tests call initialize first?

Reply:
every thing using table will need to call `table::initialize()` first, and we don't even have support for setup()..